pkg/hyperkube/hyperkube.go

```diff
@@ -164,6 +165,9 @@ func (hk *HyperKube) Run(args []string, stopCh <-chan struct{}) error {
 
 	s, err := hk.FindServer(serverName)
 	if err != nil {
+		if len(args) > 0 {
+			goto RunAgain // the first args was popped off at start of Run, try again with new args
```

Comment:
How/Why does this work?

Reply:
did you read the helpful comment?

Reply:
why do we have to pop off args at all?

Reply:
That was how this code worked. I, clearly, did not write that part.

Reply:
the existing code popped off the first item in the array because it represents the entrypoint (e.g. main.go or the name of the binary) and isn't really one of the arguments for the command.

Reply:
I think I get it, we're looking for `apiserver` or `controller-manager` specifically here. Seems like there's something weird still.

thanks @carolynvs